core/src/main/java/org/apache/iceberg/ReachableFileCleanup.java

```diff
@@ -64,18 +64,11 @@ public void cleanFiles(TableMetadata beforeExpiration, TableMetadata afterExpira
       }
     }
 
-    Set<ManifestFile> candidateManifestFilesForDeletion = readManifests(expiredSnapshots);
-    Set<ManifestFile> manifestFilesAfterExpiration = readManifests(snapshotsAfterExpiration);
+    Set<ManifestFile> manifestsToDelete = readManifests(snapshotsBeforeExpiration);
```

Comment:
Why does this use `snapshotsBeforeExpiration` rather than `expiredSnapshots`? I think the ones referenced by expired snapshots are the only possible manifests to delete, and having a smaller initial set may save some work later if the new logic can short circuit.

Reply:
Ooh great catch, it was expiredSnapshots in the initial implementation but I think when shuffling code I somehow changed it to snapshotsBeforeExpiration. will fix.